### ISSUE TITLE
is this one an issue?

### ISSUE THREAD

**Issue by byman64** [monkeysuffrage/phpuri#14]
echo phpUri::parse('https://www.google.com/images/large')->join('animals/italy/tiger.jpg');
    echo "\n"; 
    // RESULT
    // => https://www.google.com/images/animals/italy/tiger.jpg
    // COMMENT
    // Is it wrong? "large" is missing...isn't it?
    echo phpUri::parse('https://www.google.com/images/large/')->join('animals/italy/tiger.jpg');  
    echo "\n"; 
    // Adding a slash to Right
    // RESULT
    // => https://www.google.com/images/large/animals/italy/tiger.jpg

**Comment by monkeysuffrage** [comment]
No, this is correct behavior.

**Comment by byman64** [comment]
Many thanks for reply and for this fantastic Class !
ciao

**Comment by byman64** [comment]
[OT] Here but I dont know how add a "Pull Request" 

It would be nice a new method on your class: getBaseURL()  (Full current URL Path) from $_SERVER
I am not a php expert, I am still learning...I found a function and I created a little classe for it.
I tried to extend your but as said it was too complex for me.

Here the class

/*

FULL URL

*/

class phpURL {
    private $s;
    public $url;

    public function __construct( $s )
    {
        if (empty($s)===true){
            $this->s=$_SERVER;
        } else {
            $this->s=$s;
        }
        $this->url=$this->getURLFull();
    }

    public function url_origin( $s, $use_forwarded_host = false ) {        
        $ssl      = ( ! empty( $s['HTTPS'] ) && $s['HTTPS'] == 'on' );
        $sp       = strtolower( $s['SERVER_PROTOCOL'] );
        $protocol = substr( $sp, 0, strpos( $sp, '/' ) ) . ( ( $ssl ) ? 's' : '' );
        $port     = $s['SERVER_PORT'];
        $port     = ( ( ! $ssl && $port=='80' ) || ( $ssl && $port=='443' ) ) ? '' : ':'.$port;
        $host     = ( $use_forwarded_host && isset( $s['HTTP_X_FORWARDED_HOST'] ) ) ? $s['HTTP_X_FORWARDED_HOST'] : ( isset( $s['HTTP_HOST'] ) ? $s['HTTP_HOST'] : null );
        $host     = isset( $host ) ? $host : $s['SERVER_NAME'] . $port;
        return $protocol . '://' . $host;
    }
    public function getURLFull( $s = false, $use_forwarded_host = false ) {
        if (empty($s)===false){
            $this->s=$s;            
        } else {
        }       

        $this->url=$this->url_origin( $this->s, $use_forwarded_host ) . $this->s['REQUEST_URI'];
        return $this->url;
    }
}

// Example 1
echo "Example 1";
$ophpURL1= new phpURL();
echo "<br>";
echo $ophpURL1->getURLFull();
echo "<br>";
echo $ophpURL1->url;

ciao

**Comment by salamonv** [comment]
I ran into this one as well, and it was unexpected to me. Why is this "expected behavior"? 
I was expecting these 2 (slash at the end, or not) to yield the same result:

echo phpUri::parse('https://www.google.com/images/large')->join('animals/italy/tiger.jpg');
echo phpUri::parse('https://www.google.com/images/large/')->join('animals/italy/tiger.jpg');

Thanks for the library!

**Comment by monkeysuffrage** [comment]
No because the base path of https://www.google.com/images/large
is https://www.google.com/images/
whereas the base path of https://www.google.com/images/large/
is https://www.google.com/images/large/

**Comment by monkeysuffrage** [comment]
No, there's no way of telling if that url is for a file or directory. A web
server would know and redirect accordingly.
